pepper-apis/src/main/java/org/broadinstitute/ddp/db/ActivityInstanceDao.java

```diff
@@ -379,6 +356,51 @@ public Long getActivityInstanceIdByGuid(Handle handle, String guid) {
         );
     }
 
+    /**
+     * Iterate through list of activity summaries and count up how many questions are answered.
+     *
+     * @param handle            the database handle
+     * @param userGuid          the user guid
+     * @param studyGuid         the study guid
+     * @param activitySummaries the list of activity summaries
+     */
+    public void countActivitySummaryQuestionsAndAnswers(Handle handle, String userGuid, String studyGuid,
+                                                        List<ActivityInstanceSummary> activitySummaries) {
+        ActivityDefStore activityDefStore = ActivityDefStore.getInstance();
+
+        Set<String> instanceGuids = activitySummaries.stream()
+                .map(ActivityInstanceSummary::getActivityInstanceGuid)
+                .collect(Collectors.toSet());
+        Map<String, FormResponse> instanceResponses = handle
+                .attach(org.broadinstitute.ddp.db.dao.ActivityInstanceDao.class)
+                .findFormResponsesWithAnswersByInstanceGuids(instanceGuids)
+                .collect(Collectors.toMap(ActivityResponse::getGuid, Function.identity()));
+
+        for (var summary : activitySummaries) {
+            if (ActivityType.valueOf(summary.getActivityType()) == FORMS) {
+                String activityCode = summary.getActivityCode();
+                String versionTag = summary.getVersionTag();
+                long versionId = summary.getVersionId();
+                long revisionStart = summary.getRevisionStart();
+
+                FormActivityDef formActivityDef = activityDefStore.getActivityDef(studyGuid, activityCode, versionTag);
+                if (formActivityDef == null) {
+                    ActivityDto activityDto = handle.attach(JdbiActivity.class)
+                            .findActivityByStudyGuidAndCode(studyGuid, activityCode).get();
+                    formActivityDef = handle.attach(FormActivityDao.class).findDefByDtoAndVersion(
+                            activityDto, versionTag, versionId, revisionStart);
+                    activityDefStore.setActivityDef(studyGuid, activityCode, versionTag, formActivityDef);
+                }
+
+                Pair<Integer, Integer> questionAndAnswerCounts = activityDefStore.countQuestionsAndAnswers(
```

Comment:
Skipping question answer counts for studies that don't need this feature should contribute  to some performance improvement.